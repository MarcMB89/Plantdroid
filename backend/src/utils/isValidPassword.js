function isValidPassword(password) {
  return password === this.password;
}
module.exports = { isValidPassword };
