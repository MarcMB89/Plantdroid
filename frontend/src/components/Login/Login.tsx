import React, {useState} from 'react';
import {
  StatusBar,
  Text,
  View,
  ImageBackground,
  Image,
  SafeAreaView,
} from 'react-native';

import {Button, TextInput} from 'react-native-paper';

import styles from '../../styles/styles';

export default function Login({navigation}: any) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const onPressHandlerRegister = () => {
    navigation.navigate('Register');
  };
  const onPressHandlerDashboard = () => {
    navigation.navigate('Dashboard');
  };

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar backgroundColor="#7BA891" barStyle="dark-content" />
      <View style={styles.background}>
        <ImageBackground
          style={styles.login_background}
          source={{uri: 'https://i.ibb.co/99S3gsz/Background-Plantdroid.jpg'}}>
          <View style={styles.login_body}>
            <Image
              source={{uri: 'https://i.ibb.co/hVqfSSp/Plantdroid-logo-v3.png'}}
              style={styles.login_logo}
            />
            <Text style={styles.login_title}>Plantdroid</Text>
            <TextInput
              style={styles.userData}
              left
              label="Email"
              value={email}
              onChangeText={item => setEmail(item)}
            />
            <TextInput
              style={styles.userData}
              left
              label="Password"
              value={password}
              secureTextEntry={true}
              onChangeText={item => setPassword(item)}
            />
            <View style={styles.login_fixToText}>
              <Button
                color="#7BA891"
                mode="contained"
                onPress={onPressHandlerRegister}>
                Register
              </Button>
              <Button
                style={styles.login_loginButton}
                color="#7BA891"
                mode="contained"
                onPress={onPressHandlerDashboard}>
                Login
              </Button>
            </View>
          </View>
        </ImageBackground>
      </View>
    </SafeAreaView>
  );
}
