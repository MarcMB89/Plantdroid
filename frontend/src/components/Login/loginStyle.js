import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({

    login_fixToText: {
        marginTop: 35,
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    
    login_loginButton: {
        marginLeft: 35
    },

    login_logo: {
        width: 130,
        height: 130,
        borderRadius: 50,
        borderColor: 'black'
    },
    
    login_body: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 50,
    },

    login_background: {
        flex: 1,
        backgroundColor: '#dcedc8'
    },

    login_title: {
        fontSize: 40,
        fontFamily: 'monospace',
        fontWeight: 'bold',
        margin: 10,
        marginTop: 25,
        marginBottom: 11,
        backgroundColor: 'white'
    },

    login_userData: {
        width: 250,
        height: 65,
        marginBottom: 5,
        marginTop: 5
    },
      
    login_container: {
        flex: 1,
    },

})

export default styles;