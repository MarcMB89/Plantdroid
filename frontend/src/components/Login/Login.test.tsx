import React from 'react';
import {render} from '../../utils/test.utils';
import Login from './Login';

let navigation: any = {
  navigate: jest.fn(),
};

describe('Given a Login Component', () => {
  describe('When it is rendered', () => {
    let screen: any;
    beforeEach(() => {
      screen = render(<Login navigation={navigation} />);
    });
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});

navigation = {navigation};
