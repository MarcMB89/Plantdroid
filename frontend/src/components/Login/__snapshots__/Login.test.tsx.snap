// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Login Component When it is rendered Then screen should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#dcedc8",
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://i.ibb.co/99S3gsz/Background-Plantdroid.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 50,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://i.ibb.co/hVqfSSp/Plantdroid-logo-v3.png",
            }
          }
          style={
            Object {
              "borderColor": "black",
              "borderRadius": 50,
              "height": 130,
              "width": 130,
            }
          }
        />
        <Text
          style={
            Object {
              "backgroundColor": "white",
              "fontFamily": "monospace",
              "fontSize": 40,
              "fontWeight": "bold",
              "margin": 10,
              "marginBottom": 11,
              "marginTop": 25,
            }
          }
        >
          Plantdroid
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(231, 231, 231)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              Object {
                "marginBottom": 5,
                "marginTop": 5,
                "width": 250,
              },
            ]
          }
        >
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.26)",
                "bottom": 0,
                "height": 2,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": Array [
                  Object {
                    "scaleY": 0.5,
                  },
                ],
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "minHeight": 65,
                },
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <Text
                collapsable={false}
                nativeID="animatedComponent"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 32,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
              <Text
                collapsable={false}
                nativeID="animatedComponent"
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 32,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
            </View>
            <TextInput
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#6200ee"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "zIndex": 1,
                  },
                  Object {
                    "paddingLeft": 12,
                    "paddingRight": 12,
                  },
                  Object {
                    "height": 65,
                  },
                  Object {
                    "paddingBottom": 4,
                    "paddingTop": 24,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(231, 231, 231)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              Object {
                "marginBottom": 5,
                "marginTop": 5,
                "width": 250,
              },
            ]
          }
        >
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.26)",
                "bottom": 0,
                "height": 2,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": Array [
                  Object {
                    "scaleY": 0.5,
                  },
                ],
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "minHeight": 65,
                },
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <Text
                collapsable={false}
                nativeID="animatedComponent"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 32,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
              <Text
                collapsable={false}
                nativeID="animatedComponent"
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 32,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
            </View>
            <TextInput
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              secureTextEntry={true}
              selectionColor="#6200ee"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "zIndex": 1,
                  },
                  Object {
                    "paddingLeft": 12,
                    "paddingRight": 12,
                  },
                  Object {
                    "height": 65,
                  },
                  Object {
                    "paddingBottom": 4,
                    "paddingTop": 24,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 35,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#7BA891",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 2,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0.75,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Register
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#7BA891",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 2,
                "marginLeft": 35,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0.75,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
